vermouth/molecule.py

```diff
@@ -674,6 +673,38 @@ def edges_between(self, n_bunch1, n_bunch2, data=False):
                 else:
                     yield (node1, node2, self.edges[node1, node2])
 
+    def _remove_interactions_with_node(self, node):
+        """
+        We iterate through the different interactions we have and
+        remove the interactions where the atoms to be deleted are present.
+        Further we also delete the entire interaction_type if it is
+        empty after all the necessary interactions have been deleted.
+        """
+        for name, interactions in self.interactions.items():
+            for interaction in interactions:
+                if node in interaction.atoms:
+                    self.interactions[name].remove(interaction)
+
+        for i in self.interactions.copy():
```

Comment:
We only need a copy of the keys, not the whole dict; and `i` is not a very good variable name.
```
for interaction_type in list(self.interactions):
    if not self.interactions[interaction_type]:
        del self.interactions[interaction_type]
```